apiutils/task.go

```diff
@@ -0,0 +1,21 @@
+package apiutils
```

Comment:
`utils` packages are usually problematic.

However, this comparison is specific to the agent. The fields selected for the comparison are relevant there. I'm worried about binding these together.